totp/totp.go

```diff
@@ -0,0 +1,60 @@
+package totp
+
+import (
+	"encoding/base32"
+	"time"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/bip32"
+	"github.com/pquerna/otp"
+	"github.com/pquerna/otp/totp"
+)
+
+// Time-base one-time password service
+type Service struct {
```

Comment:
I would suggest other name for this struct - not sure what exactly, but imo Service is "default"

Reply:
What is wrong with "default" name? Isn't it enough informative with package name `totp.Service` ?

Reply:
So are you saying that an entity which provides some functionality but is not initialized when its client is created is not a service?

Also, I see two reasons why getting xpriv from the client is incorrect:
* it puts unnecessary limitations on the TOTP service, restricting its ability to generate passwords for only one key. Why?
* it forces the correct order of `ClientOpts` during client creation

Reply:
To be honest I was confused during the review because of the name `totp` 🙈 . I realized that we have:
- `totp.go` file in root directory 
- `totp/totp.go` file which holds some kind of algorithm & utils functions which utilizes:
- `totp` external lib (to be exact "github.com/pquerna/otp/totp")  
- ...and of course:
- `totp/totp_test.go` 
- and the package `totp` inside `/totp` directory

My humble suggestion is...maybe:
- `totp.go` in root - leave it as is
- move `totp/totp.go` to root dir with a name e.g. `totp_generator.go`
- In the `totp_generator.go`:
   - get rid of the `Service` and make `GenerateTotp` and `ValidateTotp` as normal functions:
   
```golang
func GenerateTotp(period uint, digits uint, xPriv *bip32.ExtendedKey, pubKey *bec.PublicKey) (string, error)
ValidateTotp(period uint, digits uint, xPriv *bip32.ExtendedKey, pubKey *bec.PublicKey, passcode string) (bool, error)
```

Reply:
I've removed totp service at all - @chris-4chain indeed it looks much better now